lms/djangoapps/mobile_api/users/serializers.py

```diff
@@ -34,6 +35,10 @@ def to_representation(self, course_overview):
                 kwargs={'course_id': course_id},
                 request=request
             )
+            discussion_url = reverse(
+                'discussion_course',
+                kwargs={'course_id': course_id},
+                request=request) if course_overview.is_discussion_tab_enabled() else None
```

Comment:
Try this:

1. `./manage.py lms --settings test test lms/djangoapps/mobile_api/users/tests.py:TestUserEnrollmentApi.test_discussion_url`

and then:
2.  `./manage.py lms --settings test test lms/djangoapps/mobile_api/users/tests.py:TestUserEnrollmentApi`

#1 *will* work.  But #2 will *not* work.  
Why?  
Because #1 correctly enables the Discussions feature flag BEFORE `lms/urls.py` is executed.  
While with #2, the Discussion feature flag is NOT enabled before `lms/urls.py` is executed.  

The fix is simple:
Move `@patch.dict("django.conf.settings.FEATURES", {"ENABLE_DISCUSSION_SERVICE": True})` to the top of the `TestUserEnrollmentApi` class.